File: python/412.py
class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        def checker(num: int) -> str:
            res_str = str(num)

            if num % 3 == 0:
                res_str = 'Fizz'
            if num % 5 == 0:
                res_str = 'Buzz'
            if num % 3 == 0 and num % 5 == 0:
                res_str = 'FizzBuzz'

            return res_str

        res = []

        for i in range(1, n + 1):
            res.append(checker(i))

        return res
    

File: javascript/412.js
/**
 * @param {number} n
 * @return {string[]}
 */
var fizzBuzz = function(n) {
    let res_arr = [];

    for (i = 1; i <= n; i++) {
        let temp_str = ''

        if (i % 3 == 0) {
            temp_str += 'Fizz'
        }
        if (i % 5 == 0) {
            temp_str += 'Buzz'
        }
        if (temp_str == '') {
            temp_str += [i].toString()
        }
        res_arr.push(temp_str)
    }
    return res_arr
};